@zi-m, thanks for the PR!
The CI system doesn't test ogre-next features, I tested them manually,  csharp, java and python build successfully . however, d3d9 failed. could you help have a look?

./vcpkg install ogre-next[d3d9]:x64-windows
Failed in cmake configure step. Failures:
CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
DirectX9_D3DX9_LIBRARY (ADVANCED)
    linked by target "RenderSystem_Direct3D9" in directory F:/VCPKG/8677/vcpkg/buildtrees/ogre-next/src/c710e36bc0-9cfa542442/RenderSystems/Direct3D9

--- a/ports/ogre-next/environment-overrides.cmake
+++ b/ports/ogre-next/environment-overrides.cmake
@@ -1,0 +1,16 @@
+#[[
+
+Building steps for ogre-next[d3d9]:
+
+1. Download and install "Microsoft DirectX SDK"
+https://www.microsoft.com/en-us/download/confirmation.aspx?id=6812
+
+2. Set env variable
+set DXSDK_DIR=C:/Program Files (x86)/Microsoft DirectX SDK (June 2010)
+
+3. Install port
+.\vcpkg.exe install ogre-next[d3d9] --triplet x64-windows
+
+]]
+
+set(VCPKG_ENV_PASSTHROUGH DXSDK_DIR)

